README.md

```diff
@@ -34,12 +34,21 @@ Extend @ni in the ESLint configuration.
 
 ### TypeScript
 
-Extend @ni/eslint-config/typescript in the ESLint configuration. Configure the @typescript-eslint plugin and the project's TypeScript configuration.
+Extend `@ni/eslint-config/typescript` in the default ESLint configuration. The default `.eslintrc.json` used for development should only extend TypeScript rules and not the TypeScript rules requiring type checking.
 
 ```json
 {
     "extends": "@ni/eslint-config/typescript",
-    "plugins": ["@typescript-eslint"],
+}
+```
+
+### TypeScript Rules Requiring Type Checking
+
+Extend `@ni/eslint-config/typescript-requiring-type-checking` in a separate ESLint configuration. Configure the project's TypeScript configuration. Due to the longer execution time, the separate `.eslintrc-requires-type-checking.json` should only run from an explicit `npm run lint` command invoked by developers and on CIs.
```

Comment:
Think we should recommend a git pre commit hook or a git pre push hook here? Might decrease developer frustration when forgetting to run before pushing to PRs for PR checks to run. Could increase frustration storing changes on temporary branches. I haven't really used them myself.

Reply:
Discussed in person. General consensus is that it would be annoying to prevent local commits due to type errors.